### ISSUE TITLE
Terraform can't count to 4

### ISSUE THREAD

**Issue by michalmedvecky** [hashicorp/terraform#15611]
### Terraform Version

Terraform v0.9.11

### Terraform Configuration Files
```hcl
resource "openstack_networking_network_v2" "net-backend" {
  name = "${var.backend_network}"
  admin_state_up = "true"
}

resource "openstack_networking_subnet_v2" "net-backend-subnet" {
  name = "${var.backend_network}"
  network_id = "${openstack_networking_network_v2.net-backend.id}"
  cidr = "${var.backend_network_cidr}"
  dns_nameservers = [ "100.127.70.70" ]
  ip_version = 4
  depends_on = ["openstack_networking_network_v2.net-backend"]
}

resource "openstack_networking_port_v2" "test_port_backend" {
  name = "test${count.index+1}"
  count = "2"
  admin_state_up = "true"
  network_id = "${openstack_networking_network_v2.net-backend.id}"
}

resource "openstack_compute_instance_v2" "test" {
  name = "test${count.index+1}"
  count = "2"
  image_id = "66ae409c-f2ea-4a8c-a25c-3859d79f9f33"
  flavor_name = "m1.small"
  key_pair = "misko"
  availability_zone = "mz"

  network {
    port = "${element(openstack_networking_port_v2.test_port_backend.*.id, count.index)}"
  }

}

variable "backend_network" {
  description = "local network name"
}

variable "backend_network_cidr" {
  description = "local network cidr"
}
```

### Expected Behavior
Plan shows `4` resources are going to be created

### Actual Behavior
Plan shows `8` resources are going to be created 
 
### Steps to Reproduce

I've been hitting this issue "randomly" on my large IaaC definition but I managed to narrow down to a reportable issue.

1. create ^ TF file(s) as above (I have them split into multiple `.tf` files but that shouldn't matter)
2. do terraform plan:
```
+ openstack_compute_instance_v2.test.0
    access_ip_v4:             "<computed>"
    access_ip_v6:             "<computed>"
    all_metadata.%:           "<computed>"
    availability_zone:        "mz"
    flavor_id:                "<computed>"
    flavor_name:              "m1.small"
    force_delete:             "false"
    image_id:                 "66ae409c-f2ea-4a8c-a25c-3859d79f9f33"
    image_name:               "<computed>"
    key_pair:                 "misko"
    name:                     "test1"
    network.#:                "1"
    network.0.access_network: "false"
    network.0.fixed_ip_v4:    "<computed>"
    network.0.fixed_ip_v6:    "<computed>"
    network.0.floating_ip:    "<computed>"
    network.0.mac:            "<computed>"
    network.0.name:           "<computed>"
    network.0.port:           "${element(openstack_networking_port_v2.test_port_backend.*.id, count.index)}"
    network.0.uuid:           "<computed>"
    region:                   "RegionOne"
    security_groups.#:        "<computed>"
    stop_before_destroy:      "false"

+ openstack_compute_instance_v2.test.1
    access_ip_v4:             "<computed>"
    access_ip_v6:             "<computed>"
    all_metadata.%:           "<computed>"
    availability_zone:        "mz"
    flavor_id:                "<computed>"
    flavor_name:              "m1.small"
    force_delete:             "false"
    image_id:                 "66ae409c-f2ea-4a8c-a25c-3859d79f9f33"
    image_name:               "<computed>"
    key_pair:                 "misko"
    name:                     "test2"
    network.#:                "1"
    network.0.access_network: "false"
    network.0.fixed_ip_v4:    "<computed>"
    network.0.fixed_ip_v6:    "<computed>"
    network.0.floating_ip:    "<computed>"
    network.0.mac:            "<computed>"
    network.0.name:           "<computed>"
    network.0.port:           "${element(openstack_networking_port_v2.test_port_backend.*.id, count.index)}"
    network.0.uuid:           "<computed>"
    region:                   "RegionOne"
    security_groups.#:        "<computed>"
    stop_before_destroy:      "false"

+ openstack_networking_network_v2.net-backend
    admin_state_up: "true"
    name:           "hubudb1-net-sc-tenant-secu-sandbox"
    region:         "RegionOne"
    shared:         "<computed>"
    tenant_id:      "<computed>"

+ openstack_networking_port_v2.test_port_backend.0
    admin_state_up:          "true"
    all_fixed_ips.#:         "<computed>"
    allowed_address_pairs.#: "<computed>"
    device_id:               "<computed>"
    device_owner:            "<computed>"
    mac_address:             "<computed>"
    name:                    "test1"
    network_id:              "${openstack_networking_network_v2.net-backend.id}"
    region:                  "RegionOne"
    security_group_ids.#:    "<computed>"
    tenant_id:               "<computed>"

+ openstack_networking_port_v2.test_port_backend.1
    admin_state_up:          "true"
    all_fixed_ips.#:         "<computed>"
    allowed_address_pairs.#: "<computed>"
    device_id:               "<computed>"
    device_owner:            "<computed>"
    mac_address:             "<computed>"
    name:                    "test2"
    network_id:              "${openstack_networking_network_v2.net-backend.id}"
    region:                  "RegionOne"
    security_group_ids.#:    "<computed>"
    tenant_id:               "<computed>"

+ openstack_networking_subnet_v2.net-backend-subnet
    allocation_pools.#:         "<computed>"
    cidr:                       "100.127.172.0/24"
    dns_nameservers.#:          "1"
    dns_nameservers.1907041309: "100.127.70.70"
    enable_dhcp:                "true"
    gateway_ip:                 "<computed>"
    ip_version:                 "4"
    name:                       "hubudb1-net-sc-tenant-secu-sandbox"
    network_id:                 "${openstack_networking_network_v2.net-backend.id}"
    region:                     "RegionOne"
    tenant_id:                  "<computed>"


Plan: 6 to add, 0 to change, 0 to destroy.
```
So far, that's correct

3. Import pre-existing net/subnet resources
```
ubuntu@deployer:~/devel/tf-bug/p$ terraform import openstack_networking_network_v2.net-backend d3ebd619-3f2d-45cb-8e28-86da7cdf4cc5
openstack_networking_network_v2.net-backend: Importing from ID "d3ebd619-3f2d-45cb-8e28-86da7cdf4cc5"...
openstack_networking_network_v2.net-backend: Import complete!
  Imported openstack_networking_network_v2 (ID: d3ebd619-3f2d-45cb-8e28-86da7cdf4cc5)
openstack_networking_network_v2.net-backend: Refreshing state... (ID: d3ebd619-3f2d-45cb-8e28-86da7cdf4cc5)

Import success! The resources imported are shown above. These are
now in your Terraform state. Import does not currently generate
configuration, so you must do this next. If you do not create configuration
for the above resources, then the next `terraform plan` will mark
them for destruction.
ubuntu@deployer:~/devel/tf-bug/p$ terraform import openstack_networking_subnet_v2.net-backend-subnet 365fb63b-ea7d-4d31-b471-eb75b686c13b
openstack_networking_subnet_v2.net-backend-subnet: Importing from ID "365fb63b-ea7d-4d31-b471-eb75b686c13b"...
openstack_networking_subnet_v2.net-backend-subnet: Import complete!
  Imported openstack_networking_subnet_v2 (ID: 365fb63b-ea7d-4d31-b471-eb75b686c13b)
openstack_networking_subnet_v2.net-backend-subnet: Refreshing state... (ID: 365fb63b-ea7d-4d31-b471-eb75b686c13b)

Import success! The resources imported are shown above. These are
now in your Terraform state. Import does not currently generate
configuration, so you must do this next. If you do not create configuration
for the above resources, then the next `terraform plan` will mark
them for destruction.
```

4. Do `terraform plan` again:
```
+ openstack_compute_instance_v2.test.0
    access_ip_v4:             "<computed>"
    access_ip_v6:             "<computed>"
    all_metadata.%:           "<computed>"
    availability_zone:        "mz"
    flavor_id:                "<computed>"
    flavor_name:              "m1.small"
    force_delete:             "false"
    image_id:                 "66ae409c-f2ea-4a8c-a25c-3859d79f9f33"
    image_name:               "<computed>"
    key_pair:                 "misko"
    name:                     "test1"
    network.#:                "1"
    network.0.access_network: "false"
    network.0.fixed_ip_v4:    "<computed>"
    network.0.fixed_ip_v6:    "<computed>"
    network.0.floating_ip:    "<computed>"
    network.0.mac:            "<computed>"
    network.0.name:           "<computed>"
    network.0.port:           "${element(openstack_networking_port_v2.test_port_backend.*.id, count.index)}"
    network.0.uuid:           "<computed>"
    region:                   "RegionOne"
    security_groups.#:        "<computed>"
    stop_before_destroy:      "false"

+ openstack_compute_instance_v2.test.1
    access_ip_v4:             "<computed>"
    access_ip_v6:             "<computed>"
    all_metadata.%:           "<computed>"
    availability_zone:        "mz"
    flavor_id:                "<computed>"
    flavor_name:              "m1.small"
    force_delete:             "false"
    image_id:                 "66ae409c-f2ea-4a8c-a25c-3859d79f9f33"
    image_name:               "<computed>"
    key_pair:                 "misko"
    name:                     "test2"
    network.#:                "1"
    network.0.access_network: "false"
    network.0.fixed_ip_v4:    "<computed>"
    network.0.fixed_ip_v6:    "<computed>"
    network.0.floating_ip:    "<computed>"
    network.0.mac:            "<computed>"
    network.0.name:           "<computed>"
    network.0.port:           "${element(openstack_networking_port_v2.test_port_backend.*.id, count.index)}"
    network.0.uuid:           "<computed>"
    region:                   "RegionOne"
    security_groups.#:        "<computed>"
    stop_before_destroy:      "false"

+ openstack_networking_port_v2.test_port_backend.0
    admin_state_up:          "true"
    all_fixed_ips.#:         "<computed>"
    allowed_address_pairs.#: "<computed>"
    device_id:               "<computed>"
    device_owner:            "<computed>"
    mac_address:             "<computed>"
    name:                    "test1"
    network_id:              "d3ebd619-3f2d-45cb-8e28-86da7cdf4cc5"
    region:                  "RegionOne"
    security_group_ids.#:    "<computed>"
    tenant_id:               "<computed>"

+ openstack_networking_port_v2.test_port_backend.1
    admin_state_up:          "true"
    all_fixed_ips.#:         "<computed>"
    allowed_address_pairs.#: "<computed>"
    device_id:               "<computed>"
    device_owner:            "<computed>"
    mac_address:             "<computed>"
    name:                    "test2"
    network_id:              "d3ebd619-3f2d-45cb-8e28-86da7cdf4cc5"
    region:                  "RegionOne"
    security_group_ids.#:    "<computed>"
    tenant_id:               "<computed>"


Plan: 8 to add, 0 to change, 0 to destroy.
```

^^^ You see only 4 resources are going to be added but Terraform reports 8

5. `terraform apply` creates 4 resources and reports it created 4 resources (correct)

**Comment by gtmtech** [comment]
#14905

**Comment by apparentlymart** [comment]
Hi @michalmedvecky! Sorry for the incorrect output here, and thanks for reporting this.

This issue has been fixed in master and the fix will be included in the 0.10.0 release once we get through the release candidate cycle that's going on right now. More details are in #14905.
